### PR TITLE
Dicomviewer

### LINKED ISSUE #207: Responsive Design for the Dicom Viewer.
The set of Tools need to be centered in the screen for smaller screen sizes.

**Ideal:**

Tools need to be centered

**Current:**

Unresponsive_Design

### PR DESCRIPTION
**Pull request by Zoey-dev**
What does this PR do?

A branch was created and the Dicomviewer component was worked on. The Tools bar was centralised to the middle of the page and the font-size of the icons were reduced to fit properly on smaller screens. Resolves #207 

List of task completed

-Centralised tools bar
-Reduced the font size of the icons

### CODE REVIEW

File: src/components/dicomViewer/DcmHeader/DcmHeader.scss [hunk 1]
```diff
@@ -16,11 +16,12 @@ $headerBlue: #73bcf7;
     margin: 0.3rem;
     svg {
       color: #fff;
+      font-size: 0.6rem;
     }
   }
   .dicom-header__body-tools,
   .dicom-header__body-resets {
-    padding-right: 4rem;
+    padding-right: 2rem;
```
Comment (jennydaman) [inline src/components/dicomViewer/DcmHeader/DcmHeader.scss:24]: Hard-coded centering is not responsive

File: .env [hunk 2]
```diff
@@ -1,4 +1,4 @@
 BROWSER=chrome
-REACT_APP_CHRIS_UI_URL="http://localhost:8000/api/v1/"
+REACT_APP_CHRIS_UI_URL="https://cube.outreachy.chrisproject.org/api/v1/"
```
[the remark this answers is not in the document]
Comment (Zoey-dev) [reply .env:2]: Okay, i will remove that

File: src/components/dicomViewer/DcmHeader/DcmHeader.scss (hunk 1 above)
Comment (Zoey-dev) [reply src/components/dicomViewer/DcmHeader/DcmHeader.scss]: @jennydaman i have made adjustments to that 
please help me review

Comment (Zoey-dev) [reply src/components/dicomViewer/DcmHeader/DcmHeader.scss]: Also, what is the least screen size we are working with. So i know how to adjust the other components.

File: .env (hunk 2 above)
[the remark this answers is not in the document]
Comment (jennydaman) [reply .env:2]: The `.env` file was not reverted

File: .gitignore [hunk 3]
```diff
@@ -34,3 +34,4 @@ yarn-error.log*
 .npm/
 .cache/
 .yarnrc
+.env
```
Comment (jennydaman) [inline .gitignore:37]: I would actually agree that this is a good thing to put in `.gitignore` but it breaks precedent, so for now it's probably better to revert this change.

**Review by jennydaman — CHANGES_REQUESTED** [review]
Using `flexbox` for centering is a good choice. But there were a few regressions.

On small screen (~780px wide) the bottom row of icons is off-center.

screenshot bottom row off center

Somewhere around ~769px there's a breakpoint which causes the bottom row of icons to become right-justified.

screenshot bottom row becomes right-justified

I am also asking @PintoGideon for comment, what do you think about smaller icons?

File: .env (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zoey-dev) [reply .env:2]: I added it to the gitignore file. Do you mean i should revert it back to what it was ?

File: .gitignore (hunk 3 above)
Comment (Zoey-dev) [reply .gitignore:37]: Okay okay

Comment (Zoey-dev) [reply .gitignore:37]: Done

**Comment by Zoey-dev** [comment]
> Using `flexbox` for centering is a good choice. But there were a few regressions.
> 
> On small screen (~780px wide) the bottom row of icons is off-center.
> 
> screenshot bottom row off center
> 
> Somewhere around ~769px there's a breakpoint which causes the bottom row of icons to become right-justified.
> 
> screenshot bottom row becomes right-justified
> 
> I am also asking @PintoGideon for comment, what do you think about smaller icons?

Okay first of all, i was thinking of using a media query to handle the positions and maybe font size of the icons below at different screen sizes. Then i think reducing the font size of the icons too would work

File: src/components/dicomViewer/DcmHeader/DcmHeader.scss (hunk 1 above)
Comment (PintoGideon) [reply src/components/dicomViewer/DcmHeader/DcmHeader.scss]: The minimum width ChRIS_ui is aiming to support is 992px throughout the app. Most of our current users are on laptops and wider screens.

**Comment by Zoey-dev** [comment]
> > Using `flexbox` for centering is a good choice. But there were a few regressions.
> > On small screen (~780px wide) the bottom row of icons is off-center.
> > screenshot bottom row off center
> > Somewhere around ~769px there's a breakpoint which causes the bottom row of icons to become right-justified.
> > screenshot bottom row becomes right-justified
> > I am also asking @PintoGideon for comment, what do you think about smaller icons?
> 
> Okay first of all, i was thinking of using a media query to handle the positions and maybe font size of the icons below at different screen sizes. Then i think reducing the font size of the icons too would work

Also the other texts should have smaller fonts too so they don't over flow on the image that much

File: src/components/dicomViewer/DcmHeader/DcmHeader.scss (hunk 1 above)
Comment (Zoey-dev) [reply src/components/dicomViewer/DcmHeader/DcmHeader.scss]: Okay 992px. I will work with that.

**Comment by PintoGideon** [comment]
Hey @Zoey-dev , I would keep the font-size of the toolbar in the bottom panel (the play, pause, stop) consistent with the one's in the header panel. Reducing the font-size for smaller widths work.

**Comment by Zoey-dev** [comment]
> Hey @Zoey-dev , I would keep the font-size of the toolbar in the bottom panel (the play, pause, stop) consistent with the one's in the header panel. Reducing the font-size for smaller widths work. We can also reduce the font-size of the overlay text in a media query.

OK then. Let me implement that

**Review by PintoGideon — COMMENTED** [review]
Changes to Implement:

1. Make the font-size of the bottom panel consistent with the header.
2. Reduce the font-size of the overlay text for smaller screen sizes
3. Have the bottom panel centered for smaller screen sizes, as opposed to right-justified.

**Comment by Zoey-dev** [comment]
> Changes to Implement:
> 
> 1. Make the font-size of the bottom panel consistent with the header.
> 2. Reduce the font-size of the overlay text for smaller screen sizes
> 3. Have the bottom panel centered for smaller screen sizes, as opposed to right-justified.

@PintoGideon, @jennydaman please review current changes

**Review by PintoGideon — CHANGES_REQUESTED** [review]
Hey @Zoey-dev ,

One small nitpick, The font-sizes for the icons for larger screens are too small. I made some edits to the `DcmHeader.scss` file to see if can get the font-sizes to be more responsive. Also, there was some small scope for cleanup.  

I made the similar edit to the `button` in the `GalleryToolbar.scss` file to make the font-sizes more responsive. You can compare our `.scss` files and share your thoughts ([my branch](https://github.com/PintoGideon/ChRIS_ui/tree/Zoey-dev-dicomviewer)). Overall, good job. The design looks much much better across all screens. 



```
@import '../../../assets/scss/helpers/mixins.scss';

$imgViewerBackgroundColour: black;
$textColor: white;
$GreyTextColor: #868686;
$opacityLevel: 0.75;
$headerBlue: #73bcf7;
$iconfont-size-md: 0.6rem;
$iconfont-size-lg:0.8rem;

.dicom-header {
  z-index: 2;
  background: rgba(0, 0, 0, $opacityLevel); //$imgViewerBackgroundColour;
  color: $textColor;
  display: flex;
  flex-wrap: nowrap;
  flex-direction: row;
  justify-content: center;

  .dicom-logo{
    position: absolute;
    left: 0;
    top: 0;
    padding: 0.8rem;
    margin-right: 4rem;
    
    &__text{
      @include media-query(1200px){
        font-size: 0.8rem;
      }

      @include media-query(1920px){      
       font-size:1.4rem;
      }
      
      color: $headerBlue !important;
      font-weight: 900;
      display: block;
    }
  }

  button {
    margin: 0.4rem -4px;
    svg {
      color: #fff;

      @include media-query(1200px){
        font-size: $iconfont-size-md;
      }

      @include media-query(1920px){      
       font-size:$iconfont-size-lg;
      }
    }
  }
  
  .dicom-header__center-tools{
    display: flex;
    flex-flow: row nowrap;

    .dicom-header__tools-text{
      display: flex;
      justify-content: center;
      align-items: center;

      @include media-query(1200px){
        font-size: $iconfont-size-lg;
      }

      @include media-query(1920px){      
       font-size: 1rem;
      }
    
    }
  }

  .pf-m-active {
    color: red !important;
  }

  button:hover,
  button:focus {
    svg {
      color: $headerBlue;
    }
  }
}

.dicomTag {
  &__list {
    &__item {
      font-family: monospace;
       &--name {
         font-weight:bold;
         
       }
    }
  }
}
```

In the GalleryToolbar.scss, I tried this:

```
  button {
      vertical-align: middle;
      svg {
        color: #fff;
       
        @include media-query(1200px){
          font-size: $iconfont-size-md;
        }
  
        @include media-query(1920px){      
         font-size:$iconfont-size-lg;
        }
      }
    }
  ```

**Comment by Zoey-dev** [comment]
> $iconfont-size-md: 0.6rem;
> $iconfont-size-lg:0.8rem;

Okay this looks even better. The icons get bigger on smaller screens and smaller on larger screens. Cool stuff. 
I'll just make effect the changes and push again. 
Thank you

**Comment by Zoey-dev** [comment]
@jennydaman  @PintoGideon hey, are there more issues here to work on. i've made another push please review

**Review by PintoGideon — APPROVED** [review]
Looks good to me.
